plugins/lsp/conform-nvim.nix

```diff
@@ -36,97 +116,122 @@ in
-      # Unknown plugin default
-      description = "Custom formatters and changes to built-in formatters";
+      formatters =
+        defaultNullOpts.mkAttrsOf types.anything { }
+          "Custom formatters and changes to built-in formatters.";
     };
-  };
 
-  config =
-    let
-      setupOptions =
-        with cfg;
-        {
-          formatters_by_ft = formattersByFt;
-          format_on_save =
-            if builtins.isAttrs formatOnSave then
-              {
-                lsp_fallback = formatOnSave.lspFallback;
-                timeout_ms = formatOnSave.timeoutMs;
-              }
-            else
-              helpers.mkRaw formatOnSave;
-          format_after_save =
-            if builtins.isAttrs formatAfterSave then
-              { lsp_fallback = formatOnSave.lspFallback; }
-            else
-              helpers.mkRaw formatAfterSave;
-          log_level = logLevel;
-          notify_on_error = notifyOnError;
-          inherit formatters;
-        }
-        // cfg.extraOptions;
-    in
-    mkIf cfg.enable {
-      extraPlugins = [ cfg.package ];
+  settingsExample = {
+    formattersByFt = {
+      bash = [
+        "shellcheck"
+        "shellharden"
+        "shfmt"
+      ];
+      cpp = [ "clang_format" ];
+      javascript = {
+        __unkeyed-1 = "prettierd";
+        __unkeyed-2 = "prettier";
+        timeout_ms = 2000;
+        stop_after_first = true;
+      };
+      "_" = [
+        "squeeze_blanks"
+        "trim_whitespace"
+        "trim_newlines"
+      ];
+    };
+    format_on_save = # Lua
+      ''
+        function(bufnr)
+          if vim.g.disable_autoformat or vim.b[bufnr].disable_autoformat then
+            return
+          end
+
+          if slow_format_filetypes[vim.bo[bufnr].filetype] then
+            return
+          end
 
-      extraConfigLua = ''
-        require("conform").setup(${helpers.toLuaObject setupOptions})
+          local function on_format(err)
+            if err and err:match("timeout$") then
+              slow_format_filetypes[vim.bo[bufnr].filetype] = true
+            end
+          end
+
+          return { timeout_ms = 200, lsp_fallback = true }, on_format
+         end
       '';
+    format_after_save = # Lua
+      ''
+        function(bufnr)
+          if vim.g.disable_autoformat or vim.b[bufnr].disable_autoformat then
+            return
+          end
+
+          if not slow_format_filetypes[vim.bo[bufnr].filetype] then
+            return
+          end
+
+          return { lsp_fallback = true }
+        end
+      '';
+    log_level = "warn";
+    notify_on_error = false;
+    notify_no_formatters = false;
+    formatters = {
+      shellcheck = {
+        command = lib.getExe pkgs.shellcheck;
+      };
+      shfmt = {
+        command = lib.getExe pkgs.shfmt;
+      };
+      shellharden = {
+        command = lib.getExe pkgs.shellharden;
+      };
+      squeeze_blanks = {
+        command = lib.getExe' pkgs.coreutils "cat";
+      };
```

Comment:
This isn't ideal, remember we're passing the actual value as the example, not the expression text.

This will render in the docs as something like `command = /nix/store/XXXX-shellcheck/bin/shellcheck;`

If you want these examples you'll have to wrap the whole thing in `literalExpression`.

Reply:
Ah yeah...

Reply:
Updated to use literalExpression. I do think it provides value to show how you can utilize the property to its fullest if you want to keep these out of global path.